File: bookOfProblems/codewars.com/cpp/8 kyu Abbreviate a Two Word Name.cpp
#include <string>

std::string abbrevName(std::string name)
{
  std::string ret;
  ret.push_back(toupper(name[0]));
  ret.push_back('.');
  ret.push_back(toupper(name[name.find(' ') + 1] ));
  return ret;
}

//////////

std::string abbrevName(std::string name)
{
  name.erase(1, name.find(' ')).erase(2).insert(1, ".");
  for (auto & c: name) c = toupper(c);
  return name;
}

File: bookOfProblems/codewars.com/golang/8 kyu Abbreviate a Two Word Name.go
package kata

import (
	"strings"
)

func AbbrevName(name string) string {
	parts := strings.Split(name, " ")
	if len(parts) != 2 {
		return ""
	}

	firstInitial := strings.ToUpper(string(parts[0][0]))
	secondInitial := strings.ToUpper(string(parts[1][0]))

	initials := firstInitial + "." + secondInitial

	return initials
}

////////////////////////////////////////////////////////////////////////////////

package kata

import "strings"

func AbbrevName(name string) string{
  words := strings.Split(name, " ")
  return strings.ToUpper(string(words[0][0])) + "." + strings.ToUpper(string(words[1][0]))
}